tests/test_common.js

```diff
@@ -486,3 +487,15 @@ test('normalizeClientRequestArgs with a single callback', async t => {
   t.deepEqual(options, {})
   t.is(callback, cb)
 })
+
+test('addTimeout / removeAllTimeouts', t => {
+  const timeoutSpy = sinon.spy()
+
+  setTimeout(() => {
+    t.equal(timeoutSpy.called, false)
+    t.end()
+  }, 1)
+
+  common.addTimeout(setTimeout(timeoutSpy, 0))
+  common.removeAllTimeouts()
```

Comment:
Nock's strategy is to test as much as possible either through the public API or, when that is not possible, through the mock surface. Would it be possible to do that, maybe modeling the test after one of the tests in `test_delay` and adding it to `test_nock_lifecycle`?

(This has been discussed in a bunch of PRs but isn't really documented anywhere else; will open a PR to add it.)